python/dgl/graphbolt/impl/sampled_subgraph_impl.py

```diff
@@ -129,3 +132,38 @@ def __post_init__(self):
             ) and isinstance(
                 self.node_pairs.indices, torch.Tensor
             ), "Nodes in pairs should be of type torch.Tensor."
+
+    def __repr__(self) -> str:
+        return _sampled_subgraph_str(self, "SampledSubgraphImpl")
+
+
+def _sampled_subgraph_str(sampled_subgraph: SampledSubgraph, classname) -> str:
+    final_str = classname + "("
+
+    def _get_attributes(_obj) -> list:
```

Comment:
nit: many duplicated _get_attributes, consider moving it to python/dgl/graphbolt/utils/internal.py in a separate PR.